app/services/ToastService.php

```diff
@@ -1,43 +1,67 @@
 <?php
 
+class ToastItem
+{
+    function __construct(
+        public string $message = "",
+        public ToastLevel $level = ToastLevel::Success
+    ) {
+    }
+}
+
 class Toast
 {
-    public static ToastLevel $level = ToastLevel::Success;
-    public static string $message = "";
+    /** @var ToastItem[] */
+    public static array $toasts = [];
     static function create($message, $level = ToastLevel::Success)
     {
-        self::$message = $message;
-        self::$level = $level;
-        header("HX-Trigger-After-Settle: toastMessage");
+        self::$toasts[] = new ToastItem($message, $level);
     }
 
-    static function pop()
+    static function stash()
     {
-        $message = self::$message;
-        $level = self::$level;
-        self::$message = "";
-        self::$level = ToastLevel::Success;
-        return [$message, $level];
+        $_SESSION['toasts'] = self::$toasts;
     }
 
-    static function exists()
+    static function popStash()
     {
-        return !!self::$message;
+        self::$toasts = [...self::$toasts, ...($_SESSION['toasts'] ?? [])];
```

Comment:
Not sure to understand how this works

Reply:
This is the same as array_merge. I can change it if it makes it clearer!